### DISCUSSION TITLE
Filter request: ARRAY_SUBSET_OF

### DISCUSSION THREAD

**Discussion by shabeerak** [milvus-io/milvus#37586] [Ideas & Feature requests]
Hello, Just wanted to check with the experts here. I do not see a way to filter entities with a Field having subset/sub-array (in any order) for a given array.
Ref: https://milvus.io/docs/array_data_type.md

Specifically, looking for something like ARRAY_SUBSET_OF filter, which would be the opposite of ARRAY_CONTAINS_ALL.

Here’s how the filter would work:

- filter="ARRAY_CONTAINS_ALL(color_coord, [7, 8])": This would match all entities where color_coord includes both 7 and 8.
- filter="ARRAY_SUBSET_OF(color_coord, [7, 8, 9])": This would match entities where color_coord is a subset of [7, 8, 9], such as  [9, 7], [7], or [7, 8], but not [7, 8, 9, 10].

Wondering if this is something that is available or could be added as a feature?

**Comment by yhmo** [comment]
Wait @xiaofan-luan to comment.

**Comment by xiaofan-luan** [comment]
> Hello, Just wanted to check with the experts here. I do not see a way to filter entities with a Field having subset/sub-array (in any order) for a given array. Ref: https://milvus.io/docs/array_data_type.md
> 
> Specifically, looking for something like ARRAY_SUBSET_OF filter, which would be the opposite of ARRAY_CONTAINS_ALL.
> 
> Here’s how the filter would work:
> 
> * filter="ARRAY_CONTAINS_ALL(color_coord, [7, 8])": This would match all entities where color_coord includes both 7 and 8.
> * filter="ARRAY_SUBSET_OF(color_coord, [7, 8, 9])": This would match entities where color_coord is a subset of [7, 8, 9], such as  [9, 7], [7], or [7, 8], but not [7, 8, 9, 10].
> 
> Wondering if this is something that is available or could be added as a feature?

This is a good suggestion. If anyone want to take it that would be good

**Reply by liliu-z** [reply]
Linked the discussion in another place: https://github.com/milvus-io/milvus/discussions/6302#discussioncomment-11212180
